### ISSUE TITLE
Asset Shelf - Add set to world origin toggle like in asset browser

### ISSUE THREAD

**Issue by Draise14** [Bforartists/Bforartists#4669]

**Comment by ReinerBforartists** [comment]
:'3

![ccode](https://github.com/user-attachments/assets/97aa743c-e675-4068-975c-b28407188c91)

**Comment by ReinerBforartists** [comment]
source\blender\editors\asset\intern\asset_shelf.cc

![location](https://github.com/user-attachments/assets/2e171cbd-2b73-4425-b681-f5105f68337f)

**Comment by ReinerBforartists** [comment]
Unsuccessful operation was unsuccessful

I have no idea how to write the python prop in c. At least it doesn't crash :)

![propinc](https://github.com/user-attachments/assets/840d346e-33b9-44f6-89f1-938736cdacff)

**Comment by ReinerBforartists** [comment]
Pointers problem. I am lost.

**Comment by Blenux** [comment]
Yeah this one also fried the brain before the outliner collection colors lol

**Comment by ReinerBforartists** [comment]
I guess we have to recreate the python prop in c ...

**Comment by ReinerBforartists** [comment]
And if we add the single buttons instead of trying to add the whole prop?

**Comment by Blenux** [comment]
Yeah was trying to attempt that too, but had errors.

**Comment by ReinerBforartists** [comment]
We could ask the chatgpt oracle. Like: 

`I have a python operator with a menu entry. How can i write this operator in a c code menu?`

Spits out quite a bit example code. But am too dumb at the moment to really follow it.

**Comment by Blenux** [comment]
I added the button so far

![image](https://github.com/user-attachments/assets/63900cc6-cb20-41e2-b6d4-d99897916799)

**Comment by ReinerBforartists** [comment]
Progress \o/

**Comment by Blenux** [comment]
Yep now just to need to figure out how to connect it to the code

**Comment by Draise14** [comment]
source/blender/editors/asset/intern/asset_shelf_asset_view.cc

![Image](https://github.com/user-attachments/assets/c4a28ea6-c0e8-44b0-a180-71a9f8adf938)

This is where we would need to change how the mode changes.

**Comment by Draise14** [comment]
@Blenux how did you make the buttons? I think seeing this here, something could be built.

**Comment by Blenux** [comment]
Sadly cant remember, lost all that work when I killed my Linux Mint install unfortuantly.

**Comment by Draise14** [comment]
We have a branch with the prototype thanks to @zNightlord
